library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,943 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'certified'}
+
+DOCUMENTATION = r'''
+---
+module: bigip_gtm_dns_listener
+short_description: configures BIG-IP DNS system to answer TCP or UDP DNS requests
+description:
+  - Defines one or more listener objects to control what protocols are available for the BIG-IP DNS system to process DNS requests.
+  - BIG-IP DNS listeners allows to use TCP and UDP protocols.
+notes:
+  - Requires BIG-IP version 12.0.0 or greater
+version_added: 2.8
+options:
+  name:
+    description:
+      - Specifies the name of the DNS Listener.
+    type: str
+    required: True
+  description:
+    description:
+      - Provides brief description for DNS Listener.
+    type: str
+  address:
+    description:
+      - Specifies the IP address on which the system listens.
+    type: str
+    required: True
+  mask:
+    description:
+      - Specifies the netmask for a network listener only.
+      - Netmask clarifies whether the host bit is an actual zero or a wildcard representation.
+    type: str
+  enabled_vlans:
+    description:
+      - List of VLANs to be enabled. When a VLAN named C(all) is used, all
+        VLANs will be allowed. VLANs can be specified with or without the
+        leading partition. If the partition is not specified in the VLAN,
+        then the C(partition) option of this module will be used.
+      - This parameter is mutually exclusive with the C(disabled_vlans) parameter.
+    type: list
+    elements: str
+  disabled_vlans:
+    description:
+      - List of VLANs to be disabled. If the partition is not specified in the VLAN,
+        then the C(partition) option of this module will be used.
+      - This parameter is mutually exclusive with the C(enabled_vlans) parameters.
+    type: list
+    elements: str
+  pool:
+    description:
+      - Specifies a default pool to which the listener to automatically direct traffic.
+    type: str
+  port:
+    description:
+      - Specifies the port on which the listener listens for connections.
+      - Valid range of values is between C(0) and C(65535) inclusive.
+    type: int
+  source_port:
+    description:
+      - Specifies whether the system preserves the source port of the connection.
+    type: str
+  translate_address:
+    description:
+      - Enables or disables address translation for the listener.
+    type: bool
+  translate_port:
+    description:
+      - Enables or disables port translation.
+    type: bool
+  irules:
+    description:
+      - Specifies list of iRules to run on listener.
+      - iRules help automate the intercepting, processing, and routing of application traffic.
+      - If you want to remove existing iRules, provide an empty list value; C([]) or empty string value C('').
```

Comment:
should only be empty list, this will cause issues if you specify a string '' on argument which is a type list, with element of string, it used to work but the behavior changed, ansible expects [''] or [] rather than '', so better to just tell them to specify empty list

Reply:
Copy that - I have updated the comment as suggested.